tests/triggers/test_combining.py

```diff
@@ -62,6 +64,126 @@ def test_repr(self, timezone, serializer):
+                    "day_of_week": "mon-fri",
+                    "timezone": timezone.utc,
+                },
+                {"_class": "IntervalTrigger", "days": 4},
+                datetime(2024, 5, 13, 0, 0, 0, tzinfo=timezone.utc),
+                [
+                    datetime(2024, 5, 13, 0, 0, 0, tzinfo=timezone.utc),
+                    datetime(2024, 5, 17, 0, 0, 0, tzinfo=timezone.utc),
+                    datetime(2024, 5, 21, 0, 0, 0, tzinfo=timezone.utc),
+                    datetime(2024, 5, 29, 0, 0, 0, tzinfo=timezone.utc),
+                ],
+            ),
+            (
+                {
+                    "_class": "CalendarIntervalTrigger",
+                    "months": 1,
+                    "timezone": timezone.utc,
+                },
+                {
+                    "_class": "CronTrigger",
+                    "day_of_week": "mon-fri",
+                    "timezone": timezone.utc,
+                },
+                datetime(2024, 1, 1, 0, 0, 0, tzinfo=timezone.utc),
+                [
+                    datetime(2024, 1, 1, 0, 0, 0, tzinfo=timezone.utc),
+                    datetime(2024, 2, 1, 0, 0, 0, tzinfo=timezone.utc),
+                    datetime(2024, 3, 1, 0, 0, 0, tzinfo=timezone.utc),
+                    datetime(2024, 4, 1, 0, 0, 0, tzinfo=timezone.utc),
+                    datetime(2024, 5, 1, 0, 0, 0, tzinfo=timezone.utc),
+                    datetime(2024, 7, 1, 0, 0, 0, tzinfo=timezone.utc),
+                    datetime(2024, 8, 1, 0, 0, 0, tzinfo=timezone.utc),
+                    datetime(2024, 10, 1, 0, 0, 0, tzinfo=timezone.utc),
+                    datetime(2024, 11, 1, 0, 0, 0, tzinfo=timezone.utc),
+                ],
+            ),
+        ],
+    )
+    def test_overlapping_triggers(
+        self, left_kwargs, right_kwargs, start_time, expected_datetimes
+    ):
+        """
+        Verify that the `AndTrigger` fires at the intersection of two triggers.
+        """
+        trigger_classes = {
+            "CronTrigger": CronTrigger,
+            "IntervalTrigger": IntervalTrigger,
+            "CalendarIntervalTrigger": CalendarIntervalTrigger,
+        }
+        left_trigger_class = trigger_classes[left_kwargs.pop("_class")]
+        right_trigger_class = trigger_classes[right_kwargs.pop("_class")]
+
+        left_kwargs.update(
+            {
+                (
+                    "start_time"
+                    if left_trigger_class is not CalendarIntervalTrigger
+                    else "start_date"
+                ): start_time,
+            }
+        )
+        right_kwargs.update(
+            {
+                (
+                    "start_time"
+                    if right_trigger_class is not CalendarIntervalTrigger
+                    else "start_date"
+                ): start_time,
+            }
+        )
+
+        left_trigger = left_trigger_class(**left_kwargs)
+        right_trigger = right_trigger_class(**right_kwargs)
```

Comment:
Why do this such a roundabout way? Why not just instantiate the triggers in the parameter declaration?

Reply:
I'm so used to lazily building objects in my tests that I guess it didn't occur to me to instantiate the triggers in the `pytest` parameters😅 I'll push another quick commit shortly